bliss/data_augmentation.py

```diff
@@ -0,0 +1,117 @@
+            aug_image, aug_tile, aug_deconv = aug_shift(aug_image, aug_tile, aug_deconv)
+
+    return aug_image, aug_tile, aug_deconv
+
+
+def aug_vflip(origin_full, image, deconv_image):
+    aug_image = TF.transforms.functional.vflip(image)
+    aug_deconv = None
+    if deconv_image is not None:
+        aug_deconv = TF.transforms.functional.vflip(deconv_image)
+    image_size = image.size(2)
+    origin_full["plocs"][:, :, 0] = image_size - origin_full["plocs"][:, :, 0] - 1
+    aug_tile = origin_full.to_tile_params(4, 4).get_brightest_source_per_tile().to_dict()
+    return aug_image, aug_tile, aug_deconv
+
+
+def aug_hflip(origin_full, image, deconv_image):
+    aug_image = TF.transforms.functional.hflip(image)
+    aug_deconv = None
+    if deconv_image is not None:
+        aug_deconv = TF.transforms.functional.hflip(deconv_image)
+    image_size = image.size(2)
+    origin_full["plocs"][:, :, 1] = image_size - origin_full["plocs"][:, :, 1] - 1
+    aug_tile = origin_full.to_tile_params(4, 4).get_brightest_source_per_tile().to_dict()
+    return aug_image, aug_tile, aug_deconv
+
+
+def aug_rotate90(origin_full, image, deconv_image):
+    aug_image = TF.transforms.functional.rotate(image, 90)
+    aug_deconv = None
+    if deconv_image is not None:
+        aug_deconv = TF.transforms.functional.rotate(deconv_image, 90)
+    image_size = image.size(2)
+    plocs = origin_full["plocs"].clone()
+    origin_full["plocs"][:, :, 1] = plocs[:, :, 0]
+    origin_full["plocs"][:, :, 0] = image_size - plocs[:, :, 1] - 1
+    aug_tile = origin_full.to_tile_params(4, 4).get_brightest_source_per_tile().to_dict()
+    return aug_image, aug_tile, aug_deconv
+
+
+def aug_rotate180(origin_full, image, deconv_image):
+    aug_image = TF.transforms.functional.rotate(image, 180)
+    aug_deconv = None
+    if deconv_image is not None:
+        aug_deconv = TF.transforms.functional.rotate(deconv_image, 180)
+    image_size = image.size(2)
+    plocs = origin_full["plocs"].clone()
+    origin_full["plocs"][:, :, 1] = image_size - plocs[:, :, 1] - 1
+    origin_full["plocs"][:, :, 0] = image_size - plocs[:, :, 0] - 1
+    aug_tile = origin_full.to_tile_params(4, 4).get_brightest_source_per_tile().to_dict()
+    return aug_image, aug_tile, aug_deconv
+
+
+def aug_rotate270(origin_full, image, deconv_image):
+    aug_image = TF.transforms.functional.rotate(image, 270)
+    aug_deconv = None
+    if deconv_image is not None:
+        aug_deconv = TF.transforms.functional.rotate(deconv_image, 270)
+    image_size = image.size(2)
+    plocs = origin_full["plocs"].clone()
+    origin_full["plocs"][:, :, 1] = image_size - plocs[:, :, 0] - 1
+    origin_full["plocs"][:, :, 0] = plocs[:, :, 1]
+    aug_tile = origin_full.to_tile_params(4, 4).get_brightest_source_per_tile().to_dict()
+    return aug_image, aug_tile, aug_deconv
+
+
+def aug_shift(origin_full, image, deconv_image):
+    shift_x = random.randint(0, 3)
+    shift_y = random.randint(0, 3)
```

Comment:
Would it work to select a shift from `random.randint(-1, 2)` instead? I doubt it will make a big difference, but perhaps shifting by -1 rather than 3 seems less likely to truncate sources.